File: app/src/api/coai.ts
import { fetchAPI } from './base';

export interface RecommendedAction {
  id: string;
  type: string;
  title: string;
  description: string;
  cta_label: string;
  target_route: string;
  prefill?: Record<string, unknown>;
  requires_confirmation: boolean;
}

export interface CoAIPriority {
  id: string;
  title: string;
  description: string;
  target_route?: string;
}

export interface CoAIAlert {
  id: string;
  type: string;
  title: string;
  description: string;
  severity: 'info' | 'warning' | 'critical' | string;
  target_route?: string;
}

export interface CoAIContextSnapshot {
  total_balance: number;
  balance_currency_count: number;
  recent_transaction_count: number;
  active_budget_count: number;
  active_goal_count: number;
  active_subscription_count: number;
}

export interface CoAIBriefResponse {
  generated_at: string;
  currency: string;
  brief: string;
  priorities: CoAIPriority[];
  alerts: CoAIAlert[];
  recommended_actions: RecommendedAction[];
  context_snapshot: CoAIContextSnapshot;
}

export interface CoAIPreferences {
  user_id: string;
  preferred_currency: string;
  focus_areas: string[];
  weekly_brief_enabled: boolean;
  proactive_alerts_enabled: boolean;
  updated_at: string;
}

export const coai = {
  getBrief: () =>
    fetchAPI<CoAIBriefResponse>('/coai/brief'),

  getPreferences: () =>
    fetchAPI<{ preferences: CoAIPreferences }>('/coai/preferences'),

  updatePreferences: (preferences: Partial<CoAIPreferences>) =>
    fetchAPI<{ preferences: CoAIPreferences }>('/coai/preferences', {
      method: 'PUT',
      body: JSON.stringify(preferences),
    }),
};

File: backend/internal/model/coai.go
package model

import "time"

type RecommendedAction struct {
	ID                   string                 `json:"id"`
	Type                 string                 `json:"type"`
	Title                string                 `json:"title"`
	Description          string                 `json:"description"`
	CTALabel             string                 `json:"cta_label"`
	TargetRoute          string                 `json:"target_route"`
	Prefill              map[string]interface{} `json:"prefill,omitempty"`
	RequiresConfirmation bool                   `json:"requires_confirmation"`
}

type CoAIPriority struct {
	ID          string `json:"id"`
	Title       string `json:"title"`
	Description string `json:"description"`
	TargetRoute string `json:"target_route,omitempty"`
}

type CoAIAlert struct {
	ID          string `json:"id"`
	Type        string `json:"type"`
	Title       string `json:"title"`
	Description string `json:"description"`
	Severity    string `json:"severity"`
	TargetRoute string `json:"target_route,omitempty"`
}

type CoAIContextSnapshot struct {
	TotalBalance            float64 `json:"total_balance"`
	BalanceCurrencyCount    int     `json:"balance_currency_count"`
	RecentTransactionCount  int     `json:"recent_transaction_count"`
	ActiveBudgetCount       int     `json:"active_budget_count"`
	ActiveGoalCount         int     `json:"active_goal_count"`
	ActiveSubscriptionCount int     `json:"active_subscription_count"`
}

type CoAIBriefResponse struct {
	GeneratedAt        time.Time           `json:"generated_at"`
	Currency           string              `json:"currency"`
	Brief              string              `json:"brief"`
	Priorities         []CoAIPriority      `json:"priorities"`
	Alerts             []CoAIAlert         `json:"alerts"`
	RecommendedActions []RecommendedAction `json:"recommended_actions"`
	ContextSnapshot    CoAIContextSnapshot `json:"context_snapshot"`
}

type CoAIPreferences struct {
	UserID                 string    `json:"user_id"`
	PreferredCurrency      string    `json:"preferred_currency"`
	FocusAreas             []string  `json:"focus_areas"`
	WeeklyBriefEnabled     bool      `json:"weekly_brief_enabled"`
	ProactiveAlertsEnabled bool      `json:"proactive_alerts_enabled"`
	UpdatedAt              time.Time `json:"updated_at"`
}

type UpdateCoAIPreferencesRequest struct {
	PreferredCurrency      *string   `json:"preferred_currency,omitempty"`
	FocusAreas             *[]string `json:"focus_areas,omitempty"`
	WeeklyBriefEnabled     *bool     `json:"weekly_brief_enabled,omitempty"`
	ProactiveAlertsEnabled *bool     `json:"proactive_alerts_enabled,omitempty"`
}
